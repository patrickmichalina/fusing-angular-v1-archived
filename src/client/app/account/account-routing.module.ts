import { AccountComponent } from './account.component'
import { NgModule } from '@angular/core'
import { RouterModule } from '@angular/router'
import { LoggedInGuard } from '../shared/services/guards/logged-in.guard.service'

@NgModule({
  imports: [
    RouterModule.forChild([
      {
        path: '',
        component: AccountComponent,
        canActivate: [LoggedInGuard],
        data: {
          meta: {
            title: 'Account'
          }
        }
      }
    ])
  ],
  exports: [RouterModule]
})
export class AccountRoutingModule {}
