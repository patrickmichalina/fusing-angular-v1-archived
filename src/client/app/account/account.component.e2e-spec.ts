import { baseUrl as url, browser } from '../../../../tools/test/jest.e2e-setup'

describe('Account Page', () => {
  test.skip('should have title', async () => {
    const title = await browser
      .goto(`${url}`)
      .cookies.clearAll()
      .cookies.set({
        url,
        name: 'id-token',
        value: 'authenticatedUserToken'
      })
      .cookies.set({
        url,
        name: 'access-token',
        value: 'B418yhwBbRTK98jPwjh4yPGhFwZRSdzz'
      })
      .goto(`${url}/account`)
      .evaluate(() => document.title)
      .end()

    expect(title).toContain('Account')
  })
})
