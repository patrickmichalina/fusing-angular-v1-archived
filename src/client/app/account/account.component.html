<div *ngIf="user$ | async as user">
  <img [src]="user.picture">
  <p>Name: <strong>{{ user.name }}</strong></p>
  <p>Nickname: <strong>{{ user.nickname }}</strong></p>
  <p>Email: <strong>{{ user.email }}</strong></p>
  <p>Email Verified: <strong>{{ user.email_verified }}</strong></p>
  <p>Updated At: <strong>{{ user.updated_at | date }}</strong></p>
</div>