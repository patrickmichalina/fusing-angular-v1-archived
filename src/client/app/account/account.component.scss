:host {
  display: block;
  img {
    height: 100px;
    width: 100px;
    max-width: 100%;
  }
}
