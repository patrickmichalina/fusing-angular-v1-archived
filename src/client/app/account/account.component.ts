import { ChangeDetectionStrategy, Component } from '@angular/core'
import { AuthService } from '../shared/services/auth.service'

@Component({
  selector: 'pm-account',
  templateUrl: './account.component.html',
  styleUrls: ['./account.component.css'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class AccountComponent {
  constructor(private auth: AuthService) {}
  readonly user$ = this.auth.user$
}
