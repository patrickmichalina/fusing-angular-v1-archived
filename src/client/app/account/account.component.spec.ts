import { AccountComponent } from './account.component'
import { async, ComponentFixture, TestBed } from '@angular/core/testing'
import { Component } from '@angular/core'
import { AccountModule } from './account.module'
import { AppTestingModule } from '../../../testing/app-testing.module'

@Component({
  selector: 'test-component',
  template: '<pm-account></pm-account>'
})
class TestComponent {}

describe(AccountComponent.name, () => {
  let fixture: ComponentFixture<TestComponent>

  beforeEach(
    async(() => {
      TestBed.configureTestingModule({
        imports: [AccountModule, AppTestingModule.forRoot()],
        declarations: [TestComponent]
      }).compileComponents()
    })
  )

  beforeEach(
    async(() => {
      fixture = TestBed.createComponent(TestComponent)
    })
  )

  afterEach(
    async(() => {
      TestBed.resetTestingModule()
    })
  )

  it(
    'should compile',
    async(() => {
      fixture.detectChanges()
      expect(fixture.componentInstance).toBeDefined()
      expect(fixture).toMatchSnapshot()
    })
  )
})
