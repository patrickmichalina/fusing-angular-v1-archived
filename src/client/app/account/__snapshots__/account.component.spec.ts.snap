// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountComponent should compile 1`] = `
<test-component>
  <pm-account>
    
    <mat-card
      class="mat-card"
    >
      <img
        src="null"
      />
      <p>
        Name: 
        <strong>
          
        </strong>
      </p>
      <p>
        Nickname: 
        <strong>
          
        </strong>
      </p>
      <p>
        Email: 
        <strong>
          
        </strong>
      </p>
      <p>
        Email Verified: 
        <strong>
          
        </strong>
      </p>
      <p>
        Updated At: 
        <strong>
          
        </strong>
      </p>
    </mat-card>
  </pm-account>
</test-component>
`;
